### PR TITLE
Add Gflag to enable/disable incremental backup feature

### PR DESCRIPTION
**Pull request by qiyanghe1998**
This diff adds a gflag to enable/disable incremental backup feature.
This is the first step for incremental backup.

For testing, I log some info to show the gflag is enabled.

Mahmoud's comment: we have not done a design review on the thread / stop mechanism, so he suggest me to hold off any change apart from the flag.

Thus, I clean my code in the last commit.

### CODE REVIEW

File: rocksdb_admin/admin_handler.h [hunk 1]
```diff
@@ -203,6 +203,9 @@ class AdminHandler : virtual public AdminSvIf {
 
   std::unique_ptr<std::thread> db_deletion_thread_;
   std::atomic<bool> stop_db_deletion_thread_;
+
+  std::unique_ptr<std::thread> db_incremental_backup_thread_;
```
Comment (mradwan) [inline rocksdb_admin/admin_handler.h:207]: Add/create these when you are going to use them so remove for now.

File: rocksdb_admin/admin_handler.cpp [hunk 2]
```diff
@@ -703,7 +723,7 @@ bool AdminHandler::backupDBHelper(const std::string& db_name,
   if (!status.ok()) {
     e->errorCode = AdminErrorCode::DB_ADMIN_ERROR;
     e->message = status.ToString();
-    LOG(ERROR) << "Error happened when opending db for backup: " << e->message;
+    LOG(ERROR) << "Error happened when opening db for backup: " << e->message;
```
Comment (mradwan) [inline rocksdb_admin/admin_handler.cpp:726]: Thank you!

File: rocksdb_admin/admin_handler.cpp [hunk 3]
```diff
@@ -447,7 +457,8 @@ AdminHandler::AdminHandler(
   , meta_db_(OpenMetaDB())
   , allow_overlapping_keys_segments_()
   , num_current_s3_sst_downloadings_(0)
-  , stop_db_deletion_thread_(false) {
+  , stop_db_deletion_thread_(false)
```
Comment (mradwan) [inline rocksdb_admin/admin_handler.cpp:460]: Only add these when you are planning to use them

File: rocksdb_admin/admin_handler.cpp [hunk 4]
```diff
@@ -495,6 +510,11 @@ AdminHandler::~AdminHandler() {
     stop_db_deletion_thread_ = true;
     db_deletion_thread_->join();
   }
+
+  if (FLAGS_enable_async_incremental_backup_dbs) {
+    stop_db_incremental_backup_thread_ = true;
+    db_incremental_backup_thread_->join();
+  }
```
Comment (mradwan) [inline rocksdb_admin/admin_handler.cpp:517]: Remove

File: rocksdb_admin/admin_handler.cpp [hunk 5]
```diff
@@ -134,6 +134,16 @@ DEFINE_int32(async_delete_dbs_wait_sec,
...
+DEFINE_bool(enable_async_incremental_backup_dbs, false, "Enable incremental backup for db files");
+
+DEFINE_int32(async_incremental_backup_dbs_frequency_sec,
+             60,
+             "How frequently in sec to check the dbs need deleting in async way");
+             
+DEFINE_int32(async_incremental_backup_dbs_wait_sec,
+             60,
+             "How long in sec to wait between the dbs deletion");
+
```
Comment (mradwan) [inline rocksdb_admin/admin_handler.cpp:146]: Remove

**Review by mradwan — COMMENTED** [review]
Lets only add code when we are ready to use ? apart from the flag?

**Comment by neildge** [comment]
looks good to me. Thank you for cleaning this up!

**Review by neildge — APPROVED** [review]
lg2m thank you for cleaning this up

**Review by jaricftw — APPROVED** [review]

**Pull request merged**